### PR TITLE
Update ion-schema and ion-rs dependencies

### PR DESCRIPTION
**Pull request by popematt**
### Issue #, if available:

None.

### Description of changes:

Updates `ion-rs` and `ion-schema` crates to the latest versions. There's also a bunch of clippy fixes that ended up here too.

----
_**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**_

### CODE REVIEW

File: src/bin/ion/commands/generate/model.rs [hunk 1]
```diff
@@ -436,6 +433,7 @@ mod model_tests {
                         ),
                     ]
                     .into_iter(),
+                    true,
```
[the remark this answers is not in the document]
Comment (jobarr-amzn) [reply src/bin/ion/commands/generate/model.rs:436]: What is the `true` here? Can you comment it or give it a name? Random booleans in interfaces grow like rust as codebases age, and inhibit readability and maintainability unless you stay on top of it, imo.

File: src/bin/ion/commands/generate/model.rs [hunk 2]
```diff
@@ -496,6 +491,7 @@ mod model_tests {
...
```
Comment (jobarr-amzn) [inline src/bin/ion/commands/generate/model.rs:494]: Same thing- what's the `true`?

File: src/bin/ion/commands/schema/validate.rs [hunk 3]
```diff
@@ -113,46 +106,27 @@ impl IonCliCommand for ValidateCommand {
...
             match validation_result {
                 Ok(_) => {
-                    writer.write_string("Valid")?;
-                    writer.set_field_name("value");
-                    writer.write_string(element_to_string(&owned_element)?)?;
-                    writer.set_field_name("schema");
-                    writer.write_string(schema_id)?;
+                    struct_writer.write("result", "Valid")?;
+                    struct_writer.write("value", format!("{}", &owned_element))?;
+                    struct_writer.write("schema", schema_id)?;
```
Comment (jobarr-amzn) [inline src/bin/ion/commands/schema/validate.rs:120]: `struct_writer` really cleans this up, nice :)

**Review by jobarr-amzn — APPROVED** [review]

File: src/bin/ion/commands/schema/validate.rs (hunk 3 above)
Comment (popematt) [reply src/bin/ion/commands/schema/validate.rs:120]: Yes, @zslayton put a lot of effort into trying different approaches in order to find an ergonomic writer API.

File: src/bin/ion/commands/generate/model.rs (hunk 2 above)
Comment (popematt) [reply src/bin/ion/commands/generate/model.rs:494]: This is part of the code gen model. It is something to do with whether the fields are closed for a given type. I'll create an issue to replace it with an enum.

File: src/bin/ion/commands/generate/model.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (popematt) [reply src/bin/ion/commands/generate/model.rs:436]: Opened https://github.com/amazon-ion/ion-schema-rust/issues/224

**Pull request merged**